compiler/ballerina-lang/src/main/resources/grammar/BallerinaParser.g4

```diff
@@ -188,11 +192,12 @@ typeName
     |   LEFT_PARENTHESIS typeName RIGHT_PARENTHESIS                         # groupTypeNameLabel
     |   LEFT_PARENTHESIS typeName (COMMA typeName)* RIGHT_PARENTHESIS       # tupleTypeNameLabel
     |   OBJECT LEFT_BRACE objectBody RIGHT_BRACE                            # objectTypeNameLabel
-    |   RECORD LEFT_BRACE fieldDefinitionList RIGHT_BRACE                   # recordTypeNameLabel
+    |   RECORD LEFT_BRACE recordFieldDefinitionList RIGHT_BRACE             # openRecordTypeNameLabel
+    |   SEALED RECORD LEFT_BRACE recordFieldDefinitionList RIGHT_BRACE      # sealedRecordTypeNameLabel
```

Comment:
Can't we bring these two rules to one? eg:
```
SEALED? RECORD LEFT_BRACE recordFieldDefinitionList RIGHT_BRACE
```

Reply:
Although we initially discussed about merging these to one, later we decided to keep these separate and change the logic in the BLangParserListener to make use of this separation.